### PR TITLE
[TS migration] Migrate 'EmojiPicker' component to TypeScript

### PR DESCRIPTION
**Pull request by lancer-spaid**
### Details

Migrate these files from js to tsx:
[src/components/EmojiPicker/getSkinToneEmojiFromIndex.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/getSkinToneEmojiFromIndex.js)	
[src/components/EmojiPicker/EmojiSkinToneList.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiSkinToneList.js)	
[src/components/EmojiPicker/EmojiPickerButtonDropdown.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerButtonDropdown.js)	
[src/components/EmojiPicker/EmojiPickerButton.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerButton.js)	
[src/components/EmojiPicker/EmojiPicker.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPicker.js)	
[src/components/EmojiPicker/CategoryShortcutButton.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/CategoryShortcutButton.js)	
[src/components/EmojiPicker/CategoryShortcutBar.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/CategoryShortcutBar.js)
[src/components/EmojiPicker/EmojiPickerMenuItem/index.native.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenuItem/index.native.js)	
[src/components/EmojiPicker/EmojiPickerMenuItem/index.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenuItem/index.js)
[src/components/EmojiPicker/EmojiPickerMenu/index.native.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenu/index.native.js)	
[src/components/EmojiPicker/EmojiPickerMenu/index.js](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenu/index.js)	

### Fixed Issues

$ [#25155](https://github.com/Expensify/App/issues/25155)
PROPOSAL: [#25155 (comment)](https://github.com/Expensify/App/issues/25155#issuecomment-1860918764)

### Tests

1. Go to any chat you will see emojis, select any emoji.
2. Press add reaction button.

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for the expected offline behavior in the `Offline steps` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [ ] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] I verified that if a function's arguments changed that all usages have also been updated correctly
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 1]
```diff
@@ -190,6 +192,6 @@ const EmojiPicker = forwardRef((props, ref) => {
     );
 });
 
-EmojiPicker.propTypes = propTypes;
 EmojiPicker.displayName = 'EmojiPicker';
```
Comment (lancer-spaid) [inline src/components/EmojiPicker/EmojiPicker.tsx:195]: `Property 'displayName' does not exist on type '(props: EmojiPickerProps & RefAttributes<unknown>) => ReactElement<any, string | JSXElementConstructor<any>> | null'`
I am facing this issue over here, how can this be dealt?

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 2]
```diff
@@ -1,67 +1,77 @@
...
-    const [activeID, setActiveID] = useState();
-    const emojiPopoverAnchorRef = useRef(null);
+    const [emojiPopoverAnchorOrigin, setEmojiPopoverAnchorOrigin] = useState<AnchorOrigin>(DEFAULT_ANCHOR_ORIGIN);
+    const [activeID, setActiveID] = useState<string | null>(null);
+    const emojiPopoverAnchorRef = useRef<View>(null) as MutableRefObject<View | null>;
     const onModalHide = useRef(() => {});
-    const onEmojiSelected = useRef(() => {});
-    const emojiSearchInput = useRef();
+    const onEmojiSelected = useRef<OnEmojiSelected>();
+    const emojiSearchInput = useRef<View>();
```
Comment (lancer-spaid) [inline src/components/EmojiPicker/EmojiPicker.tsx:48]: This part may need some further considerations...

File: src/components/EmojiPicker/EmojiPicker.tsx (hunk 1 above)
Comment (lancer-spaid) [reply src/components/EmojiPicker/EmojiPicker.tsx:195]: Hi @abdulrahuman5196 please look at this as this is causing some issue in the component. Do you have an idea?

Comment (abdulrahuman5196) [reply src/components/EmojiPicker/EmojiPicker.tsx:195]: I think its due to usage of forwardRef here https://github.com/Expensify/App/pull/34162/files#diff-eba20b923a2357dff4d39f91a106832801db71ad38932473581c84db7cc52e7dR35

Can you try changing the  use of forwardRef in the export line?

Comment (abdulrahuman5196) [reply src/components/EmojiPicker/EmojiPicker.tsx:195]: @lancer-spaid

File: src/libs/actions/EmojiPickerAction.ts [hunk 3]
```diff
@@ -66,7 +67,7 @@ function showEmojiPicker(
 /**
  * Hide the Emoji Picker modal.
  */
-function hideEmojiPicker(isNavigating: boolean) {
+function hideEmojiPicker(isNavigating: boolean ) {
```
Comment (blazejkustra) [inline src/libs/actions/EmojiPickerAction.ts:70]: 👀

File: src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import * as Emojis from '@assets/emojis';
+import type { OnyxKeyValue } from '@src/ONYXKEYS';
+
+/**
+ * Fetch the emoji code of selected skinTone
+ * @returns {String}
+ */
```
Comment (blazejkustra) [inline src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx:7]:
```suggestion
```

File: src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+type SkinToneIndex = OnyxKeyValue<"preferredEmojiSkinTone">;
+
+type SkinToneEmoji = {
+    readonly code: string;
+    readonly skinTone: number;
+}
+
```
Comment (blazejkustra) [inline src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx:15]:
```suggestion
/** Fetch the emoji code of selected skinTone */
```

File: src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+function getSkinToneEmojiFromIndex(skinToneIndex: SkinToneIndex) {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/getSkinToneEmojiFromIndex.tsx:16]:
```suggestion
function getSkinToneEmojiFromIndex(skinToneIndex: SkinToneIndex): string {
```

File: src/components/EmojiPicker/EmojiSkinToneList.tsx [hunk 7]
```diff
@@ -11,22 +10,23 @@ import CONST from '@src/CONST';
...
         setIsSkinToneListVisible((prev) => !prev);
     }, []);
-
-    /**
-     * Set the preferred skin tone in Onyx and close the skin tone picker
-     * @param {object} skinToneEmoji
-     */
-    function updateSelectedSkinTone(skinToneEmoji) {
+    
+    function updateSelectedSkinTone(skinToneEmoji: SkinToneEmoji) {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiSkinToneList.tsx:29]: Let's leave descriptions:
```suggestion
/** Set the preferred skin tone in Onyx and close the skin tone picker */
    function updateSelectedSkinTone(skinToneEmoji: SkinToneEmoji) {
```

File: src/components/EmojiPicker/EmojiSkinToneList.tsx [hunk 8]
```diff
@@ -11,22 +10,23 @@ import CONST from '@src/CONST';
...
 
+type SkinToneEmoji = {
+    readonly code: string;
+    readonly skinTone: number;
+}
+
 function EmojiSkinToneList() {
     const styles = useThemeStyles();
-    const [highlightedIndex, setHighlightedIndex] = useState(null);
+    const [highlightedIndex, setHighlightedIndex] = useState(-1);
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiSkinToneList.tsx:20]: Please make sure it doesn't break anything

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 9]
```diff
@@ -75,18 +83,14 @@ function EmojiPickerButtonDropdown(props) {
...
     <EmojiPickerButtonDropdown
         // eslint-disable-next-line react/jsx-props-no-spreading
         {...props}
         forwardedRef={ref}
     />
 ));
 
-EmojiPickerButtonDropdownWithRef.displayName = 'EmojiPickerButtonDropdownWithRef';
-
 export default withLocalize(EmojiPickerButtonDropdownWithRef);
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:96]: Use useLocalize hook instead

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 10]
```diff
@@ -75,18 +83,14 @@ function EmojiPickerButtonDropdown(props) {
...
 EmojiPickerButtonDropdown.displayName = 'EmojiPickerButtonDropdown';
 
-const EmojiPickerButtonDropdownWithRef = React.forwardRef((props, ref) => (
+const EmojiPickerButtonDropdownWithRef = React.forwardRef((props: EmojiPickerButtonDropdownProps, ref: React.ForwardedRef<View>) => (
     <EmojiPickerButtonDropdown
         // eslint-disable-next-line react/jsx-props-no-spreading
         {...props}
         forwardedRef={ref}
     />
 ));
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:94]: Remove this part and instead do it like in guidelines:
```ts
- [1.2](#forwardRef) **`forwardRef`**

  ```ts
  // CustomTextInput.tsx

  import { forwardRef, useRef, ReactNode, ForwardedRef } from "react";
  import { TextInput, View } from "react-native";

  export type CustomTextInputProps = {
    label: string;
    children?: ReactNode;
  };

  function CustomTextInput(props: CustomTextInputProps, ref: ForwardedRef<TextInput>) {
    return (
      <View>
        <TextInput ref={ref} />
        {props.children}
      </View>
    );
  };

  export default forwardRef(CustomTextInput);
```

(rename forwardedRef to ref)

```suggestion
```

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 11]
```diff
@@ -58,8 +65,9 @@ function EmojiPickerButtonDropdown(props) {
                     <View style={styles.emojiPickerButtonDropdownContainer}>
                         <Text
                             style={styles.emojiPickerButtonDropdownIcon}
-                            numberOfLines={1}
+                            numberOfLines={2}
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:68]: Why this change?

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 12]
```diff
@@ -79,10 +89,8 @@ const EmojiPicker = forwardRef((props, ref) => {
 
     /**
      * Hide the emoji picker menu.
-     *
-     * @param {Boolean} isNavigating
      */
-    const hideEmojiPicker = (isNavigating) => {
+    const hideEmojiPicker = (isNavigating: boolean) => {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPicker.tsx:93]:
```suggestion
    const hideEmojiPicker = (isNavigating = false) => {
```

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 13]
```diff
@@ -1,38 +1,45 @@
...
+function EmojiPickerButtonDropdown({isDisabled = false, ...props}: EmojiPickerButtonDropdownProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const emojiPopoverAnchor = useRef(null);
     useEffect(() => EmojiPickerAction.resetEmojiPopoverAnchor, []);
 
     const onPress = () => {
         if (EmojiPickerAction.isEmojiPickerVisible()) {
-            EmojiPickerAction.hideEmojiPicker();
+            EmojiPickerAction.hideEmojiPicker(false);
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:42]: Let's add a default value and leave it without arguments:
```suggestion
            EmojiPickerAction.hideEmojiPicker();
```

File: src/components/EmojiPicker/EmojiPickerButton.tsx [hunk 14]
```diff
@@ -68,7 +71,6 @@ function EmojiPickerButton(props) {
     );
 }
 
-EmojiPickerButton.propTypes = propTypes;
-EmojiPickerButton.defaultProps = defaultProps;
 EmojiPickerButton.displayName = 'EmojiPickerButton';
-export default compose(withLocalize, withNavigationFocus)(EmojiPickerButton);
+
+export default withLocalize(withNavigationFocus(EmojiPickerButton));
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButton.tsx:76]: Change withLocalize and withNavigationFocus  into hooks too

File: src/components/EmojiPicker/EmojiPickerButton.tsx [hunk 15]
```diff
@@ -1,37 +1,40 @@
...
 import * as EmojiPickerAction from '@userActions/EmojiPickerAction';
+import type { Emoji } from '@assets/emojis/types';
 
-const propTypes = {
+type OnEmojiSelected = (emojiCode: string, emojiObject: Emoji) => void;
+
+type EmojiSelected = {
+    emojiCode: string;
+    emojiObject: Emoji;
+}
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButton.tsx:20]:
```suggestion
}

```

File: src/components/EmojiPicker/EmojiPickerButton.tsx [hunk 16]
```diff
@@ -1,37 +1,40 @@
...
+type EmojiPickerButtonProps = WithLocalizeProps & {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPickerButton.tsx:21]: Not all params should be required here, throughout the PR please pay closer attention to prop-types (isRequired signals that param is required).

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 17]
```diff
@@ -1,67 +1,77 @@
...
     const [emojiPopoverAnchorPosition, setEmojiPopoverAnchorPosition] = useState({
         horizontal: 0,
         vertical: 0,
     });
-    const [emojiPopoverAnchorOrigin, setEmojiPopoverAnchorOrigin] = useState(DEFAULT_ANCHOR_ORIGIN);
-    const [activeID, setActiveID] = useState();
-    const emojiPopoverAnchorRef = useRef(null);
+    const [emojiPopoverAnchorOrigin, setEmojiPopoverAnchorOrigin] = useState<AnchorOrigin>(DEFAULT_ANCHOR_ORIGIN);
+    const [activeID, setActiveID] = useState<string | null>(null);
+    const emojiPopoverAnchorRef = useRef<View>(null) as MutableRefObject<View | null>;
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPicker.tsx:45]:
```suggestion
    const emojiPopoverAnchorRef = useRef<View | null>(null);
```

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 18]
```diff
@@ -1,67 +1,77 @@
...
 };
 
-const EmojiPicker = forwardRef((props, ref) => {
+type OnEmojiSelected = (emoji: string, emojiObject: Emoji) => void;
+
+type EmojiPickerProps = {
+    viewportOffsetTop: number;
+}
+
+const EmojiPicker = forwardRef(({viewportOffsetTop}: EmojiPickerProps, ref) => {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPicker.tsx:35]: Move `forwardRef` to export line, check other Ts components

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 19]
```diff
@@ -1,67 +1,77 @@
...
 const DEFAULT_ANCHOR_ORIGIN = {
     horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.RIGHT,
     vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
 };
 
-const propTypes = {
-    viewportOffsetTop: PropTypes.number.isRequired,
+type OnModalHideValue = () => void;
+
+type AnchorOrigin = {
```
Comment (blazejkustra) [inline src/components/EmojiPicker/EmojiPicker.tsx:23]: I think this is defined somewhere else, please reuse this type

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 20]
```diff
@@ -102,30 +110,24 @@ const EmojiPicker = forwardRef((props, ref) => {
...
     };
 
     /**
      * Whether emoji picker is active for the given id.
-     *
-     * @param {String} id
-     * @return {Boolean}
      */
-    const isActive = (id) => Boolean(id) && id === activeID;
+    const isActive = (id: string) => Boolean(id) && id === activeID;
```
Comment (filip-solecki) [inline src/components/EmojiPicker/EmojiPicker.tsx:130]:
```suggestion
    const isActive = (id: string) => !!id && id === activeID;
```

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 21]
```diff
@@ -1,38 +1,45 @@
...
-};
+type EmojiPickerButtonDropdownProps = WithLocalizeProps & {
+    isDisabled?: boolean;
+    value: string;
+    style: StyleProp<ViewStyle>;
+    disabled: boolean;
+    onModalHide: () => void;
+    onInputChange: (value: string) => EmojiSelected;
+    // eslint-disable-next-line react/no-unused-prop-types
+    forwardedRef?: React.ForwardedRef<View>;
```
Comment (filip-solecki) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:31]: it should not be included here, do it like in `Text` component

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx [hunk 22]
```diff
@@ -58,8 +65,9 @@ function EmojiPickerButtonDropdown(props) {
                     <View style={styles.emojiPickerButtonDropdownContainer}>
                         <Text
                             style={styles.emojiPickerButtonDropdownIcon}
-                            numberOfLines={1}
+                            numberOfLines={2}
                         >
+                            Whatever you want to do.
```
Comment (filip-solecki) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.tsx:70]: ???

File: src/components/EmojiPicker/EmojiPickerButton.tsx (hunk 14 above)
Comment (filip-solecki) [inline src/components/EmojiPicker/EmojiPickerButton.tsx:76]: use `useLocalize` instead of withLocalize

File: src/components/EmojiPicker/EmojiPicker.tsx [hunk 23]
```diff
@@ -1,67 +1,77 @@
-import PropTypes from 'prop-types';
-import React, {forwardRef, useCallback, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import type {MutableRefObject} from 'react';
+import React, { forwardRef, useCallback, useEffect, useImperativeHandle, useRef, useState} from 'react';
```
Comment (filip-solecki) [inline src/components/EmojiPicker/EmojiPicker.tsx:2]: remove unnecessary spaces in imports

**Comment by blazejkustra** [comment]
Please sync this branch with the newest main. Context [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1706196639925099), thanks!

**Comment by blazejkustra** [comment]
@lancer-spaid Bump

**Comment by blazejkustra** [comment]
@abdulrahuman5196 What happens when a contributor disappears in the middle of PR? 😅

**Comment by abdulrahuman5196** [comment]
I will raise this in the GH @blazejkustra

**Comment by abdulrahuman5196** [comment]
Raised concern in GH for alternate options if the contributor is not available - https://github.com/Expensify/App/issues/25155#issuecomment-1934773454
